app/controllers/questions_controller.rb

```diff
@@ -70,11 +183,32 @@ def log_time
   end
 
   def next
-    setup = Setup.take
-    week = setup.weeks.find_by_number(params[:week_number])
-    RecommenderSystem::Recommender.setup(current_user.id,week.id)
-    best = RecommenderSystem::HybridRecommender.new.get_best
-    los = LearningObject.find(best[0])
-    redirect_to action: "show", id: los.url_name
+
+    @room = Room.find_by_id(params[:room_number])
+
+    if !(current_user.show_solutions)
+      id_array = []
+      @results=RoomsLearningObject.get_id_do_not_viseted(params[:room_number])
+      @results.each do |r|
+        id_array.push(r['learning_object_id'].to_i)
+      end
+
+      if (id_array.empty?)
+        lo = @room.learning_objects.all.distinct
+        lo.each do |l|
+          id_array.push(l.id)
+        end
+      end
+
+      los = @room.learning_objects.find(id_array[Random.rand(0..(id_array.count-1))])
```

Comment:
Toto je uplne zbytocne - ziskat z pola IDcka, tie zamiesat cez random a potom znova podla tych IDciek zase vyhladavat.

Malo by stacit pouzit `shuffle` (a nielen na `id_array` a potom znova vyhladavat, ale rovno na to pole learning objectov).